libshaderc/include/shaderc.h

```diff
@@ -24,12 +24,26 @@ extern "C" {
 #include <stdint.h>
 
 typedef enum {
-  shaderc_glsl_vertex_shader,
-  shaderc_glsl_fragment_shader,
-  shaderc_glsl_compute_shader,
-  shaderc_glsl_geometry_shader,
-  shaderc_glsl_tess_control_shader,
-  shaderc_glsl_tess_evaluation_shader
+  // Forced shader kinds. These shader kinds force the compiler to compile the
+  // source code as the specified kind of shader.
+  shaderc_glsl_forced_vertex_shader,
```

Comment:
Another thought.  Maybe we can have both enum names.
*_infer_from_source can internally be translated to whatever set of more specific options alllow you to infer from the source, whether it's pragmas or other language features.
This could be done in a subsequent CL.
So let's say Coder's Choice for using the enum name based on pragmas.